src/events/command-handler/CorePreCommandRun.ts

```diff
@@ -13,7 +13,7 @@ export class CoreEvent extends Event<Events.PreCommandRun> {
 	public async run(message: Message, command: Command, parameters: string, commandName: string, prefix: string) {
 		const result = await command.preconditions.run(message, command);
 		if (isErr(result)) {
-			this.client.emit(Events.CommandDenied, message, command, parameters, commandName, prefix);
+			this.client.emit(Events.CommandDenied, message, command, result.error, parameters, commandName, prefix);
```

Comment:
I feel that the error should be the first argument, similar to error callbacks.

cc: @Favna @vladfrangu

Reply:
I feel like it should be consistent no matter what event we emit.. otherwise we can just make it an object

Reply:
I agree with Vlad

Reply:
Same, I'm leaning towards objects, because most of our events send a lot of different parameters.

Reply:
So what do you want me to do here?